Rubberduck.Parsing/VBA/ParseCoordinator.cs

```diff
@@ -75,6 +80,12 @@ private void ReparseRequested(object sender, EventArgs e)
             CancellationToken token;
             lock (_cancellationSyncObject)
             {
+                if (State.Status == ParserState.Busy)
```

Comment:
I think this is still the wrong point to suspend. The only point where it is really safe to to that is the very point where the run lock gets taken.

Reply:
This check is not thread safe since variable assignment is not thread safe. You could test against an integer you increment and decrement via `Interlocked` in `SuspendRequested`.

Reply:
This thing about this again, a more correct solution would be to introduce locks around the `Status` in the `RubberduckParserState`. Actually, that could probably be handled by using 'Interlocked.Exchange` when setting the `Status`.